presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -243,8 +243,8 @@ public static HiveColumnStatistics fromMetastoreApiColumnStatistics(ColumnStatis
                     OptionalDouble.empty(),
                     OptionalLong.empty(),
                     OptionalLong.empty(),
-                    OptionalLong.of(longStatsData.getNumNulls()),
-                    OptionalLong.of(longStatsData.getNumDVs()));
+                    longStatsData.isSetNumNulls() ? OptionalLong.of(longStatsData.getNumNulls()) : OptionalLong.empty(),
```

Comment:
This repeating pattern could probably be captured to a method:
```
private static OptionalLong valueIfSet(boolean isSet, long value);
```

Reply:
Skipping this. We do a lot of the same thing in this section of the code with other types too e.g. Optional<T> and OptionalDouble.  Extracting a method makes it harder to tell that they're all the same.

Reply:
i didn't quite understand the explanation, but that's not too important.